pyuvdata/uvcal/uvcal.py

```diff
@@ -663,7 +896,57 @@ def set_lsts_from_time_array(self, background=False):
             proc.start()
             return proc
 
-    def check(self, check_extra=True, run_check_acceptability=True):
+    def _check_flex_spw_contiguous(self):
+        """
+        Check if the spectral windows are contiguous for flex_spw datasets.
+
+        This checks the flex_spw_id_array to make sure that all channels for each
+        spectral window are together in one block, versus being interspersed (e.g.,
+        channel #1 and #3 is in spw #1, channels #2 and #4 are in spw #2).
+
+        """
+        if self.flex_spw:
```

Comment:
We could, I suppose. The first input won't be None, though, because the spw array is specified even in `flex_spw` is False. I think it's better to check the `flex_spw` attribute, I guess I could pass that into the util function.

Reply:
I was originally thinking about this in the context of removing as much redundant code as possible, but on second thought, it seems silly to call a function with an argument that specifies whether or not you need to have called said function. Comment withdrawn!